lcls_tools/superconducting/scLinacUtils.py

```diff
@@ -0,0 +1,123 @@
+from time import sleep
+
+from epics import PV
+from epics.ca import CASeverityException
+
+SSA_STATUS_ON_VALUE = 3
+SSA_SLOPE_LOWER_LIMIT = 0.5
+SSA_SLOPE_UPPER_LIMIT = 1.5
+
+# TODO add limits for the HL cavities
+LOADED_Q_LOWER_LIMIT = 3.895e7
+LOADED_Q_UPPER_LIMIT = 4.305e7
+DESIGN_Q_LOADED = 4.1e7
+
+CAVITY_SCALE_UPPER_LIMIT = 40
+CAVITY_SCALE_LOWER_LIMIT = 10
+
+RF_MODE_SELAP = 0
+RF_MODE_SELA = 1
+RF_MODE_SEL = 2
+RF_MODE_SEL_RAW = 3
+RF_MODE_PULSE = 4
+RF_MODE_CHIRP = 5
+
+SAFE_PULSED_DRIVE_LEVEL = 15
+NOMINAL_PULSED_ONTIME = 70
+
+STEPPER_TEMP_LIMIT = 70
+DEFAULT_STEPPER_MAX_STEPS = 1000000
+DEFAULT_STEPPER_SPEED = 20000
+MAX_STEPPER_SPEED = 60000
+
+# these values are based on the list of enum states found by probing {Magnettype}:L{x}B:{cm}85:CTRL
+MAGNET_RESET_VALUE = 10
+MAGNET_ON_VALUE = 11
+MAGNET_OFF_VALUE = 12
+MAGNET_DEGAUSS_VALUE = 13
+MAGNET_TRIM_VALUE = 1
+
+
+class PulseError(Exception):
```

Comment:
These three functions are all the same inside? I think they may be called in different scenarios?  Maybe there's a way to have a general error func that returns the type instead of many of the same errors w/ different names? We can discuss if that doesn't make sense.

Reply:
Yeah they're thrown in different scenarios and trigger different actions. I couldn't think of a better way to do it but I agree that this was is painful haha I'm not sure how to make it a function without pre defining the errors...